src/storage/src/sink/healthcheck.rs

```diff
@@ -111,64 +28,62 @@ pub enum SinkStatus {
         /// the `mz_sink_status_history` table.
         hint: Option<String>,
     },
-    /// Represents a irrecoverable failure in the pipeline. Data from this collection
-    /// is not queryable any longer. The only valid transition from Failed is Dropped.
-    Failed {
-        /// Error string used to populate the `error` column in the `mz_sink_status_history` table.
-        error: String,
-        /// Optional hint string which if present, will be added to the `details` column in
-        /// the `mz_sink_status_history` table.
-        hint: Option<String>,
-    },
-    /// Represents a sink that was dropped.
-    Dropped,
+    // Managed by the storage controller.
+    // Dropped,
```

Comment:
nit: this can be removed? Also, what does it mean that it's managed by the storage controller?

Reply:
the storage controller emits the final "dropped" status to the status shard